src/v/storage/disk_log_impl.cc

```diff
@@ -289,10 +304,11 @@ ss::future<std::optional<ss::sstring>> disk_log_impl::close() {
       && !_eviction_monitor->promise.get_future().available()) {
         _eviction_monitor->promise.set_exception(segment_closed_exception());
     }
+
```

Comment:
Not from this PR, but let's move the "closing log" line up top, before we do any awaits (I can imagine this being extra hard to debug without realizing we're even in this call). Or maybe update the text to indicate that we've waited for locks